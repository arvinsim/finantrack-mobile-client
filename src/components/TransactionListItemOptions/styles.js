import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  edit: { flex: 1 },
  delete: { flex: 1 }
})

export default styles