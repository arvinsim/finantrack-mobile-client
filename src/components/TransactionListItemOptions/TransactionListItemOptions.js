import styles from './styles.js'

import React, { Component } from 'react'
import { 
  Grid,
  Col,
  Button 
} from 'react-native-elements'

class TransactionListItemOptions extends Component {
  render() {
    const { handleEditTransaction, handleDeleteTransaction } = this.props
    return (
      <Grid>
        <Col>
          <Button buttonStyle={styles.edit} title='Edit' onPress={handleEditTransaction} />
        </Col>
        <Col>
          <Button buttonStyle={styles.delete} title='Delete' onPress={handleDeleteTransaction} />
        </Col>
      </Grid>
    )
  }
}

TransactionListItemOptions.propTypes = {
  handleEditTransaction: React.PropTypes.func.isRequired,
  handleDeleteTransaction: React.PropTypes.func.isRequired
}

export default TransactionListItemOptions