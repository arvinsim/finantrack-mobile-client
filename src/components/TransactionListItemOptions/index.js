import TransactionListItemOptions from './TransactionListItemOptions'

export default TransactionListItemOptions