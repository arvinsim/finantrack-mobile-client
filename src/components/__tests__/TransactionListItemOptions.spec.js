import 'react-native'
import React from 'react'
import TransactionListItemOptions from '../TransactionListItemOptions'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(
    <TransactionListItemOptions
      handleEditTransaction={()=>{}}
      handleDeleteTransaction={()=>{}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})