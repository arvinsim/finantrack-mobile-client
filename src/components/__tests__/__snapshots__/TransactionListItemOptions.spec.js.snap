// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  activeOpacity={1}
  flex={1}
  flexDirection="row"
>
  <View
    activeOpacity={1}
    flex={1}
    flexDirection="column"
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 15,
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Object {
              "backgroundColor": "#33691e",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Edit
        </Text>
      </View>
    </View>
  </View>
  <View
    activeOpacity={1}
    flex={1}
    flexDirection="column"
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 15,
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            Object {
              "backgroundColor": "#33691e",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
